FWCore/Framework/test/callback_t.cppunit.cc

```diff
@@ -254,6 +299,20 @@ void testCallback::sharedPtrTest() {
   call(callback);
   CPPUNIT_ASSERT(handle.get() == prod.ptr_.get());
   CPPUNIT_ASSERT(prod.ptr_->value_ == 2);
+
+  // null products
+  prod.produce_ = false;
+  callback.newRecordComing();
+  call(callback);
+  CPPUNIT_ASSERT(handle.get() == nullptr);
+
+  call(callback);
```

Comment:
Why the second `call(callback)`? Naively I'd expect the framework never to call again until `newRecordComing` has been called. Maybe it is because caching failed if the callback produces nothing?

Reply:
> I guess other older tests have multilple `call(callback)` as well.

Right. I _guess_ the idea is that if a `call(callback)` is called without `newRecordComing`, the callback itself is called, but the state of the "handle" is not changed.